Harvest register.spec

melyav92 opened the pull request.

melyav92 committed:
----------

melyav92 committed:
added save harvest report object

melyav92 committed:
added save harvest report object

melyav92 committed:
added test which should verify added data in the report
added test which should update and verify the report with new values
added test which should delete the report (should be updated)

melyav92 committed:
added delete report test

melyav92 committed:
added delete pen from the report test

melyav92 committed:
uncommented after hook

tamysko commented on cypress/support/pageobjects/harvest-register.js:
```diff
@@ -78,6 +78,50 @@ export class HarvestRegister{
         return cy.get('a.scp-registered-date.scp-harvest-date.selected').prev()
     };
+    confirmDeleteReportButton(){
+        cy.wait(500)
```
Why do you need this wait? cy.get waits for element. e.g.
// waits 10s for the element
cy.get('#some-delayed-element', { timeout: 10000 }).click();

tamysko commented on cypress/integration/tests/harvest-register.spec.js:
```diff
@@ -17,6 +17,14 @@ const fishPenAfterHarvestValueForM2 = 64000;
 const penM2Comment = "Comment for pen M2";
 const successfulToasterPopupMessage = 'Saved successfully';
 const reportDate = Cypress.moment().format("DD/MM/YYYY");
+const updatedFishHarvestedValueForPenM1 = 70000;
```
`const` is a signal that the identifier won’t be reassigned.
`let` is a signal that the variable may be reassigned. 
https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75

are you sure it should be const?

tamysko commented on cypress/integration/tests/harvest-register.spec.js:
```diff
@@ -26,14 +34,15 @@ let login = new LoginPage()
+         harvestRegister.sendDeleteReportRequestForTheCurrentDate(locationId)
+    })
 
     it('should register harvest harvest report for the current date',function (){
```
harvest 2 times

tamysko commented on cypress/integration/tests/harvest-register.spec.js:
```diff
@@ -60,8 +69,128 @@ describe('Harvest register',function (){
+            ,averageHarvestWeightValueForM2
+            ,fishPenAfterHarvestValueForM2
+            ,penM2Comment
+            ,successfulToasterPopupMessage)
```
Why "comma" at the beginning of the word? 

```suggestion
harvestRegister.addHarvestReport(penM1,
            fishHarvestedValueForPenM1,
            averageHarvestWeightValueForM1, ....
```

tamysko commented on cypress/integration/tests/harvest-register.spec.js:
```diff
@@ -60,8 +69,128 @@ describe('Harvest register',function (){
+        harvestRegister.deletePenItemFromTable(penM2).click()
+        harvestRegister.saveButton().click()
+        harvestRegister.pageDataIsLoaded()
+        harvestRegister.getPenObjectByName(penM2)
+            .should('not.exist')
```
is there any popup message we can verify as well?

tamysko commented on cypress/support/pageobjects/harvest-register.js:
```diff
@@ -78,6 +78,50 @@ export class HarvestRegister{
+        this.saveButton().click()
+        this.toasterPopup()
+            .should('have.text',successfulToasterPopupMessage)
+        this.pageDataIsLoaded()
```
Does it make sense to return the page after deletion? 
You can think about this :)

tamysko reviewed (changes requested):
I dont like waits, commas in metrod variables :).

melyav92 committed:
- updated objects names
- moved commas to the end of the word
- changed report date variable from 'const' to 'let'

melyav92 committed:
- updated objects names
- moved commas to the end of the word
- changed report date variable from 'const' to 'let'

melyav92 committed:
- removed redundant harvest word

melyav92 committed:
- moved username, password, penM1, penM2, locationId variables as env variables in the Cypress.json file
- updated tests with getting these data from the env

melyav92 committed:
refactored select current date step

melyav92 committed:
- added fixters tetst-data file
- added fishHarvestedValueForPenM1 to fixtures file

tamysko commented on cypress/support/pageobjects/harvest-register.js:
```diff
@@ -11,29 +11,31 @@ export class HarvestRegister{
     };
-    sendDeleteReportRequestForTheCurrentDate(locationId){
-        const deleteReportDate = Cypress.moment().format("YYYY-MM-DD");
+    sendDeleteReportRequest(locationId){
+        let deleteReportDate = Cypress.moment().format("YYYY-MM-DD");
         cy.request({
             method: 'DELETE',
             url: `/api/harvests/harvest?locationId=${locationId}&date=${deleteReportDate}`,
             //failOnStatusCode: false
         })
     }
```
`sendDeleteReportRequest(locationId, date = Cypress.moment().format("YYYY-MM-DD") ){
        cy.request({
            method: 'DELETE',
            url: `/api/harvests/harvest?locationId=${locationId}&date=${deleteReportDate}`,
        })`

tamysko commented on cypress/support/pageobjects/harvest-register.js:
```diff
@@ -11,29 +11,31 @@ export class HarvestRegister{
     };
-    sendDeleteReportRequestForTheCurrentDate(locationId){
-        const deleteReportDate = Cypress.moment().format("YYYY-MM-DD");
+    sendDeleteReportRequest(locationId){
+        let deleteReportDate = Cypress.moment().format("YYYY-MM-DD");
         cy.request({
             method: 'DELETE',
             url: `/api/harvests/harvest?locationId=${locationId}&date=${deleteReportDate}`,
             //failOnStatusCode: false
         })
```
```suggestion
    sendDeleteReportRequest(locationId, deleteReportDate = Cypress.moment().format("YYYY-MM-DD") ){
        cy.request({
            method: 'DELETE',
            url: `/api/harvests/harvest?locationId=${locationId}&date=${deleteReportDate}`,
            //failOnStatusCode: false
        })
```

This how it looks like, the way you did it is the same delete current date, but with different implementation. 
Read this: https://www.sitepoint.com/es6-default-parameters/

tamysko reviewed (changes requested):
Please fix sendDeleteReportRequest according to comment :)

melyav92 committed:
- refactored fixture invoking by using alias

melyav92 committed:
- refactored delete report request as requested
- moved test data to the fixtures file
- refactored tests to use test data from the fixtures file

melyav92 committed:
- refactored delete report request as requested
- moved test data to the fixtures file
- refactored tests to use test data from the fixtures file

melyav92 committed:
- refactored sen delete request object
- removed delete report date from the hooks

melyav92 committed:
- refactored getting fixtures data by using import

melyav92 committed:
- refactored getting fixtures data by using import

melyav92 committed:
--

melyav92 committed:
Merge branch 'master' into harvest-register.spec
